sway-fmt-v2/src/items/item_enum.rs

```diff
@@ -1,8 +1,78 @@
-        todo!()
+    fn format(&self, formatter: &mut Formatter) -> FormattedCode {
+        // TODO: creating this formatted_code with String::new() will likely cause lots of
+        // reallocations maybe we can explore how we can do this, starting with with_capacity may help.
+        let mut formatted_code = String::new();
+        let enum_variant_align_threshold = formatter.config.structures.enum_variant_align_threshold;
+
+        if let Some(visibility_token) = &self.visibility {
+            formatted_code.push_str(visibility_token.span().as_str());
+            formatted_code.push(' ');
+        }
+
+        // Add enum token
+        formatted_code.push_str(self.enum_token.span().as_str());
+        formatted_code.push(' ');
+
+        // Add name of the enum.
+        formatted_code.push_str(self.name.as_str());
+        handle_open_bracket(&mut formatted_code, formatter);
+
+        let type_fields = &self.fields.clone().into_inner().value_separator_pairs;
+
+        // In first iteration we are going to be collecting the lengths of the enum variants.
+        let variant_length: Vec<usize> = type_fields
+            .iter()
+            .map(|variant| variant.0.name.as_str().len())
+            .collect();
+
+        // Find the maximum length in the variant_length vector that is still smaller than enum_variant_align_threshold.
+        let mut max_valid_variant_length = 0;
+
+        variant_length.iter().for_each({
+            |length| {
+                if *length > max_valid_variant_length && *length < enum_variant_align_threshold {
+                    max_valid_variant_length = *length;
+                }
+            }
+        });
+
+        // In second iteration we are going to be comparing current variants length and maximum accepted variant length
+        // for calculating the allignment required.
+        for (index, type_field) in type_fields.iter().enumerate() {
+            let type_field = &type_field.0;
+            // Push the current indentation level
+            formatted_code.push_str(&formatter.shape.indent.to_string(formatter));
+            formatted_code.push_str(type_field.name.as_str());
+            formatted_code.push_str(" : ");
+
+            // Currently does not apply custom formatting for ty,
+            let current_variant_length = variant_length[index];
+
+            if current_variant_length < max_valid_variant_length {
+                // We need to add allignment between : and ty
+                // max_valid_variant_length: the length of the variant that we are taking as a reference to allign
+                // current_variant_length: the length of the current variant that we are trying to format
+                let required_allignment = max_valid_variant_length - current_variant_length;
+                // TODO: Improve handling this
+                formatted_code.push_str(&(0..required_allignment).map(|_| ' ').collect::<String>());
```

Comment:
here if we are starting from the first element, I'm pretty sure we can just do `..required_alignment` instead of `0..`

Reply:
Unfortunately providing just the upper limit like `..required_alignment` creates a `RangeTo` rather than a `Range` and `RangeTo` cannot be used as an iterator according to the [doc](https://doc.rust-lang.org/std/ops/struct.RangeTo.html). I couldn't find a way to use it like that but if you have any suggestions would be happy to apply